### DISCUSSION TITLE
Porting Label.Dispatcher from WPF to Avalonia

### DISCUSSION THREAD

**Discussion by NirabhraDas013** [AvaloniaUI/Avalonia#6631] [Q&A] [answered]
Hi, I have a project in wpf format which I am trying to port over in avalonia. I got most of the stuff error free. But there is something I am very much stuck with.

WPF code:
```
DownloadSpeed.Dispatcher.Invoke(() =>
            {
                DownloadSpeed.Content = string.Empty;
            });
```

I changed it in the Avalonia project like this:

```
this.FindControl<Label>("DownloadSpeed").Dispatcher.Invoke(() =>
            {
                DownloadSpeed.Content = string.Empty;
            });
```

In VS2019, The error is:
![image](https://user-images.githubusercontent.com/54432376/134474510-ade665d7-d0d2-4e17-98e3-70ab8990c26e.png)

I cannot find a solution to this. Please help

**Comment by grokys** [comment, marked as answer]
You can use:

```
Dispatcher.UIThread.Post(() =>
{
    DownloadSpeed.Content = string.Empty;
});
```

Avalonia only supports a single UI thread due to OSX limitations so we don't need a `Dispatcher` property on each control.

**Reply by NirabhraDas013** [reply]
Thank you so much :).
